### PR TITLE
feat: Updated HTML curriculum - draft

### PR DESCRIPTION
**Pull request by beaucarnes**
Updated HTML curriculum requested by @QuincyLarson.
Work in progress.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: curriculum/updated-curriculum/html/basic-html-04.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+  <body>
+    <h1>Hello World</h1>
+    <h2>CatPhotoApp</h2>
+    <p>Click here to view more cat photos.</p>
+  </body>
+</html>
+
+<!-- Commenting allows you to leave messages without affecting the output displayed to the end user. It also allows you to make code inactive. A comment in HTML starts with <!--, contians any number of lines of text, and ends with -->. Add a comment above the p element with the text: "TODO: Add link to cat photos". For example the comment <!-- TODO: Remove h1 -->contains the text "TODO: Remove h1". -->
```
Comment (QuincyLarson) [inline curriculum/updated-curriculum/html/basic-html-04.html]: We really want to make these concise and let people discover for themselves (or infer from context). Rather than explain the purpose of comments or that they can be multiple lines, or that they make the code inside them inactive, we should demonstrate this through subsequent examples. So I recommend we make the text as simple as:

You can leave comments in your code like this: . Comments in HTML start with . Add a comment above the p element with the text: "TODO: Add link to cat photos".

File: curriculum/updated-curriculum/html/basic-html-06.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+  <body>
+    <h2>CatPhotoApp</h2>
+    <!-- TODO: Add link to cat photos -->
+    <p>Click here to view more cat photos.</p>
+  </body>
+</html>
+
+<!-- HTML5 has tags that identify different content areas. These tags make your HTML easier to read, and also help with Search Engine Optimization (SEO) and accessibility. Identify the main section of this page by adding a <main> opening tag after the h2 element. Add a </main> closing tag after the p element. -->
```
Comment (QuincyLarson) [inline curriculum/updated-curriculum/html/basic-html-06.html]: I think at this point we should skip the introduction of more advanced structural elements like "main". We could introduce them in the accessibility project, since they have accessibility implications.

File: curriculum/updated-curriculum/html/basic-html-07.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  <body>
+    <h2>CatPhotoApp</h2>
+    <main>
+    <!-- TODO: Add link to cat photos -->
+    <p>Click here to view more cat photos.</p>
+	</main>
+  </body>
+</html>
+
+<!-- HTML elements are often nested within other HTML elements. On this page, a comment and a p element are nested within the main element. When you nest an element inside another element, the nested element becomes a child of the parent. To make HTML easier to read, you should indent child elements inside their parent element. Indent the comment and p element on this page so it is easer to see that they are children of the main element. (Note: Indentation, spacing and new lines do not impact how a browser renders a page but can make it easier to read HTML.) -->
```
Comment (QuincyLarson) [inline curriculum/updated-curriculum/html/basic-html-07.html]: "HTML elements are often nested within other HTML elements." I recommend avoiding passive voice like this, as it sounds boring. Also, this test text is way longer than it needs to be. 

Even though I recommend removing the <main> element, here's how I would reword this test:

File: curriculum/updated-curriculum/html/basic-html-04.html (hunk 1 above)
Comment (Ellajoke) [reply curriculum/updated-curriculum/html/basic-html-04.html:10]: 👍

**Review by Ellajoke — COMMENTED** [review]
👍

**Comment by Ellajoke** [comment]
👍
